metricbeat/mb/testing/data_generator.go

```diff
@@ -15,27 +15,28 @@ import (
 )
 
 var (
-	// To enable the data building, run go test  `github.com/elastic/beats/metricbeat/module/system/memory/... -data=true`
-	dataFlag = flag.Bool("data", false, "Enabled creating of data")
+	// Use `go test -data` to update files.
+	dataFlag = flag.Bool("data", false, "Write updated data.json files")
 )
 
-func WriteEvent(f mb.EventFetcher, t *testing.T) error {
-
+func WriteEvent(f mb.EventFetcher, t testing.TB) error {
 	if !*dataFlag {
-		t.Skip("Skip data generation tests")
+		t.Skip("skip data generation tests")
 	}
 
 	event, err := f.Fetch()
 	if err != nil {
 		return err
 	}
 
-	return createEvent(event, f)
+	fullEvent := CreateFullEvent(f, event)
+	WriteEventToDataJSON(t, fullEvent)
+	return nil
 }
 
-func WriteEvents(f mb.EventsFetcher, t *testing.T) error {
+func WriteEvents(f mb.EventsFetcher, t testing.TB) error {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function WriteEvents should have comment or be unexported